File: Farming.Backend/Farming.Domain/Entities/Plant.cs
using Farming.Domain.ValueObjects.Identity;
using Farming.Domain.ValueObjects.Plant;
using Farming.Shared.Abstractions.Domain;

namespace Farming.Domain.Entities
{
    public class Plant : Tenant
    {
        public PlantId Id { get; }
        public PlantName Name { get; }
        public PlantRequiredAmountPerHectare RequiredAmountPerHectare { get; }
        public PlantDescription Description { get; }
        public PlantUnit Unit { get; }

        public ICollection<Fertilizer> SuitableFertilizers { get; }
        public ICollection<Pesticide> SuitablePesticides { get; }
        public ICollection<PlantAction> PlantActions { get; }
        public ICollection<PlantWarehouseDelivery> PlantWarehouseDeliveries { get; }
        public ICollection<PlantWarehouseState> PlantWarehouseStates { get; }
    }
}

File: Farming.Frontend/src/app/core/models/plant.ts
export interface PlantDto {
  id: string;
  name: string;
  requiredAmountPerHectare: number;
  unit: string;
  description: string;
}

export interface PlantStateDto {
  plantId: string;
  plantName: string;
  quantity: number;
  unit: string;
  requiredAmountPerHectare: number;
  enoughForArea: number;
}

export interface PlantActionDto {
  landId: string;
  plantId: string;
  plantWarehouseId: string;
  quantity: number;
}
